lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1686,6 +1688,82 @@ class DecomposeAtenPadOp : public OpRewritePattern<AtenPadOp> {
 };
 } // namespace
 
+namespace {
+// Decompose `aten.split` op into `aten.slice` and `prim.ListConstruct` op.
+template <typename OpTy>
+class DecomposeAtenSplitLikeOp : public OpRewritePattern<OpTy> {
+public:
+  using OpRewritePattern<OpTy>::OpRewritePattern;
+  LogicalResult matchAndRewrite(OpTy op,
+                                PatternRewriter &rewriter) const override {
+    SmallVector<int64_t> offsets;
+    // TODO: Support non static offsets.
+    if (!matchPattern(op.split_sizes(), m_TorchConstantIntList(offsets)))
+      return rewriter.notifyMatchFailure(op, "constant split sizes supported");
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+
+    auto tensorType = op.self().getType().template dyn_cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    for (unsigned i = 0; i < tensorType.getSizes().size(); i++)
+      sizes.push_back(-1);
+
+    auto listType = op.getType().template cast<Torch::ListType>();
+    Location loc = op.getLoc();
+    SmallVector<int64_t> endIdx;
+    endIdx.push_back(offsets[0]);
+    for (unsigned i = 1; i < offsets.size(); i++) {
+      endIdx.push_back(offsets[i] + endIdx[i - 1]);
+    }
+    int64_t start = 0;
+    SmallVector<Value> listOfTensors;
+    Value stepVal = rewriter.create<Torch::ConstantIntOp>(
+        loc, rewriter.getI64IntegerAttr(1));
+    for (unsigned i = 0; i < offsets.size(); i++) {
```

Comment:
nit: it seems that `i` is only used to index `endIdx`, so I would do something like 

```cpp
for (int64_t end : endIdx) { ... }
```